### ISSUE TITLE
jira init fails when selecting default board for on-premise install

### ISSUE THREAD

**Issue by DannyJJK** [ankitpokhrel/jira-cli#173]
**Describe the bug**
I ran `jira init` to connect to my JIRA on-premise server, however after selecting the default board, the program panicked.

**Please provide following details**
1. JiraCLI Version:
    ```
    (Version="0.1.0", GitCommit="b2f09ec08a3d277de9e3530621572184de2ecf7a", GoVersion="go1.17.2", BuildDate="2021-10-31"", Compiler="gc", Platform="darwin/amd64")
    ```
2. Are you using Jira cloud or on-premise jira server?
    ```
    On-premise
    ```
3. What operating system are you using? Also mention version.
    ```
    macOS Monterey 12.0.1 (21A559)
    ```
4. What terminal are you using? Also mention version.
    ```
    alacritty 0.9.0 (fed349a)
    ```

**To Reproduce**

1. Export JIRA password to `JIRA_API_TOKEN` shell variable
2. Run jira init and follow prompts
3. See error after selecting default board

**Expected behavior**
I have not used jira-cli before, but I would expect it would finish setup and allow me to run further commands

**Screenshots**
![image](https://user-images.githubusercontent.com/11057501/139708459-4bd1823f-4ae6-4f6e-9864-20cff525b15d.png)

**Additional context**
This might depend on the board setup, but I'm not sure how to provide that to you. We are using a Kanban board setup
Panic information here:
```
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/ankitpokhrel/jira-cli/internal/config.(*JiraCLIConfig).configureMetadata(0xc0006fdcf8)
	/Users/ankitpokhrel/go/src/github.com/ankitpokhrel/jira-cli/internal/config/generator.go:303 +0x651
github.com/ankitpokhrel/jira-cli/internal/config.(*JiraCLIConfig).Generate(0x17c2860)
	/Users/ankitpokhrel/go/src/github.com/ankitpokhrel/jira-cli/internal/config/generator.go:79 +0x74
github.com/ankitpokhrel/jira-cli/internal/cmd/init.initialize(0xc0006b1900, {0x1855e8e, 0x0, 0x0})
	/Users/ankitpokhrel/go/src/github.com/ankitpokhrel/jira-cli/internal/cmd/init/init.go:29 +0x65
github.com/spf13/cobra.(*Command).execute(0xc0006b1900, {0x1f9deb0, 0x0, 0x0})
	/Users/ankitpokhrel/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0xc0006b1680)
	/Users/ankitpokhrel/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x3ad
main.main()
	/Users/ankitpokhrel/go/src/github.com/ankitpokhrel/jira-cli/cmd/jira/main.go:12 +0x1e
```

**Comment by ankitpokhrel** [comment]
Hi @DannyJJK

Thank you for reporting the issue. It looks like it failed when setting the custom field for `epic`. Is the epic disabled in your instance?

**Comment by DannyJJK** [comment]
I can select Epic from the Issue Type list on this board when creating a new issue, so I believe it is enabled, at least for this board.

**Comment by ankitpokhrel** [comment]
Is it possible for you to execute the curl command below and check what you get in some part of the output?

```sh
# Please replace everything in <> with actual info
curl --request GET \
  --url 'http://<server.url>/rest/api/2/issue/createmeta?projectKeys=<KEY>&expand=projects.issuetypes.fields' \
  --user "<USER>:<PASSWORD>"
```

Inside `issuetypes` key there should be an entry with `"name": "Epic Name",` in `fields`. What do you get in that section?

Also, could you please mention the jira version of your instance?

**Comment by DannyJJK** [comment]
Version:
Atlassian JIRA Project Management Software (v7.7.0#77001-sha1:3be3151)

Here is the "Epic" entry inside the `issuetypes` key. I've copied up to the end point of the `fields` section for this issue type:

```
               "self":"https://jira.mlp.agency/rest/api/2/issuetype/10000",
               "id":"10000",
               "description":"Created by JIRA Software - do not edit or delete. Issue type for a big user story that needs to be broken down.",
               "iconUrl":"https://jira.mlp.agency/images/icons/issuetypes/epic.svg",
               "name":"Epic",
               "subtask":false,
               "expand":"fields",
               "fields":{
                  "summary":{
                     "required":true,
                     "schema":{
                        "type":"string",
                        "system":"summary"
                     },
                     "name":"Summary",
                     "hasDefaultValue":false,
                     "operations":[
                        "set"
                     ]
                  },
```

**Comment by ankitpokhrel** [comment]
Just checked and the way epic is linked in Jira v7 seems bit different than v8. I will work on the fix to support v7.

**Comment by ankitpokhrel** [comment]
This is possibly fixed with #174. Could you please confirm by installing binary from `HEAD`? (requires `make`, `git` and `go`)

```sh
brew uninstall jira-cli
brew install jira-cli --HEAD
```

**Comment by DannyJJK** [comment]
After installing the binary from `HEAD` this now works correctly. After the "Creating new configuration...", it now correctly generates the config file and shows me where it's located.
